include/openssl/core_numbers.h

```diff
@@ -100,6 +101,68 @@ OSSL_CORE_MAKE_FUNC(void, OP_digest_freectx, (void *vctx))
 OSSL_CORE_MAKE_FUNC(void *, OP_digest_dupctx, (void *vctx))
 OSSL_CORE_MAKE_FUNC(size_t, OP_digest_size, (void))
 OSSL_CORE_MAKE_FUNC(size_t, OP_digest_block_size, (void))
+OSSL_CORE_MAKE_FUNC(int, OP_digest_ctrl, (void *vctx, int cmd, int p1, void *p2))
+
+#define OSSL_FUNC_DIGEST_ALLOC_METHODS(name, CTX_NAME) \
```

Comment:
I like these constructors!  However, I would prefer if the were in a different header file, so as not to make this one a complete mess.  I would also suggest to make one header file for each constructor set.  For example, the OSSL_FUNC_DIGEST constructors would be collected in `core_mkdigest.h`